rfcs/slider-component.md

```diff
@@ -0,0 +1,112 @@
+# Slider Component
+
+Represents control with sliding axis to choose a single value or range between min and max values.
+
+## Usage
+
+### Basic usage
+
+```javascript
+import * as React from 'react';
+import {Slider} from 'baseui/slider';
+
+export default () => <Slider range={[
+  {value: 0, label: '0, min'},
+  {value: 100, label: '100, max'}
+]} />;
+```
+
+### Advanced usage
+
+```javascript
+import * as React from 'react';
+import {StatefulSlider} from 'baseui/slider';
+
+export default () => {
+  return <StatefulSlider
+    range={[
+      {value: 0, label: '0, min'},
```

Comment:
I'm just going to comment because I don't know how things are done here - but to give a bit more concrete feedback, the idea of a `Point: {value: number, label: string}` with a `label` is a great example of something that increases new, non-native API surface while adding relatively little value because most of the time the label can be inferred from the `step` property.   Alteration to tick placement and labels can live in the overrides:
```
<RangeSlider
      key="my-range-slider"
      min={0}
      max={50}
      step={5}
      value={[15, 45]}
      onChange={action('onChangeValues')]}
      overrides={{
         ticksAt: (min, max, step) => ([0, 20, 40, 60, 80, 100]),
         tickFormat: (value) => (<div>(value === 0 ? `${value}, min` : value)</div>)
      }}
/>
```

Reply:
`Overrides` are optional. They are mostly used for styles and not for functional amendment. In your example we need to use them required.
By native you mean original `input type="range"`? So original inputs are very weak and insufficient, that's why we have these custom components.
Also there are lot's of sliders (in Surveys, settings, etc) that do not use `step` and need to know all the range of values, for instance:
```js
<Caption>To what extend you satisfied?</Caption>
<StatefulSlider
    range={[
      {value: 1, label: 'I am not satisfied'},
      {value: 2, label: 'I am disappointed'},
      {value: 3, label: 'Neutral'},
      {value: 4, label: 'I am satisfied'},
      {value: 5, label: 'I am very satisfied'}
    ]} />
```

Reply:
Yes, absolutely overrides should be optional, but not using them for functional aspects of the component seems like a dogma that could lead to limiting architectural choices.  I disagree that "original inputs are very weak and insufficient."  There was a lot of though that went into native component APIs and many have stood up surprisingly well over time.   I would write the particular example you describe out as:
```
<RangeSlider
      key="my-range-slider"
      min={1}
      max={5}
      step={1}
      defaultValue={3}
      onChange={action('onChangeValues')]}
      overrides={{
         tickFormat: (value) => ({1: 'I am not satisfied', 2: 'I am disappointed', ...})(value))
      }}
/>
```
This would preserve all native slider props and only overrides would be different.

Reply:
>  seems like a dogma that could lead to limiting architectural choices.
It's not a dogma. We can use it for functional change. However, default behavior and usage of any component is not intended to use `overrides`, otherwise it neglects our architecture. `input type=range` is very rare to be used unlike `input type=text`, which has become very powerful. Native range still doesn't support multiple thumbs, and ticks are supported through `<datalist>` with provided `id` reference, which is not the way of abstraction modern frameworks use. Just complying with native API would be not so necessary need to befriend it with current design requirements

Reply:
Okay so you're saying that a user will almost always want to format the label on a stepped range's tick values.  I'm not entirely sure that's the case and, even if it is, it might be best to just bump bump that `override` up to a new `prop` rather than creating a structure like `Point`: 
```
<RangeSlider
      key="my-range-slider"
      min={1}
      max={5}
      step={1}
      defaultValue={3}
      onChange={action('onChangeValue')]}
      tickFormat={(value) => ({1: 'I am not satisfied', 2: 'I am disappointed', ...})(value))}
/>
```
I wouldn't argue that you should use reproduce anything janky like I assume `<datalist />` is.  I'm just saying that API surface reuse is an awesome thing to maximize - both to make the component easier to use and to simplify the implementation.  I think it is likely that you can satisfy your design requirements while being just a bit more aligned with the `input type=range`.

Reply:
@benshope  Thanks for your feedback. The team had no majority in preferences on the API naming and other stuff, so we left it as is so far for future consideration